hypercerts/hooks/useMintHypercertStrategy.ts

```diff
@@ -0,0 +1,52 @@
+import { isAddress } from "viem";
+import { useAccount, useWalletClient } from "wagmi";
+
+import { useAccountStore } from "@/lib/account-store";
+import { useHypercertClient } from "@/hooks/use-hypercert-client";
+import { useQueueMintBlueprint } from "@/blueprints/hooks/queueMintBlueprint";
+import { useStepProcessDialogContext } from "@/components/global/step-process-dialog";
+
+import { EOAMintHypercertStrategy } from "../EOAMintHypercertStrategy";
+import { MintHypercertStrategy } from "../MintHypercertStrategy";
+import { SafeMintHypercertStrategy } from "../SafeMintHypercertStrategy";
+
+export const useMintHypercertStrategy = () => {
+  const { address, chain } = useAccount();
+  const { client } = useHypercertClient();
+  const { selectedAccount } = useAccountStore();
+  const dialogContext = useStepProcessDialogContext();
+  const queueMintBlueprint = useQueueMintBlueprint();
+  const walletClient = useWalletClient();
+
+  return (blueprintId?: number): MintHypercertStrategy => {
+    const activeAddress =
+      selectedAccount?.address || (address as `0x${string}`);
+
+    if (!activeAddress || !isAddress(activeAddress))
+      throw new Error("No address found");
+    if (!chain) throw new Error("No chain found");
+    if (!client) throw new Error("No HypercertClient found");
+    if (!walletClient) throw new Error("No walletClient found");
```

Comment:
nit: the SDK will throw when a write method is called without a wallet client. As the code is now it might be more legible, but `walletClient` in app doesn't mean `walletClient` in `hypercertClient`

Reply:
So would you remove this check here and let the SDK throw?

Reply:
Closing this thread to get this feature shipped today. GainForest have asked for it already.